### ISSUE TITLE
A table becomes in read-only mode when trying to flush the buffer during the async insert

### ISSUE THREAD

**Issue by tanchik-curly** [ClickHouse/ClickHouse#57455]
Hi, I tried to switch to an async insert to avoid db overloading with many small inserts. Still, I faced a problem that I started to receive a lot of 408 errors (on query level I'm doing axios query inside a cluster with max execution time config from 9 to 15 seconds, depending on query).
I tried to increase async_insert_busy_timeout_ms to 5000/4000/2000, but it became even worse, I checked the asynchronous_insert_log table and saw that some queries take minutes to be executed, what should I improve then?

![image](https://github.com/ClickHouse/ClickHouse/assets/67906338/e861cc75-061d-4d57-89f1-7691d195c069)
![image](https://github.com/ClickHouse/ClickHouse/assets/67906338/f971e60e-60b6-4ed5-9244-4c9a1b345f28)
![image](https://github.com/ClickHouse/ClickHouse/assets/67906338/eb966936-94ef-46c0-9855-9cfef2736435)

**Comment by lampjian** [comment]
Try to increase `async_insert_busy_timeout_ms` to 30000 of half a minute.

**Comment by tanchik-curly** [comment]
@lampjian but does it mean that all servers will wait half a minute for an acknowledge? because I have wait_for_async_insert = 1, but my system should process everything fast as I can receive 100k messages per 10 minutes

**Comment by lampjian** [comment]
No, it means the longest time you can see the latest data. 100k is very small, a reasonable size of batches is 1000k+. You can learn more about [asynchronous-insert mechanism](https://clickhouse.com/docs/en/optimize/asynchronous-inserts).
